File: Services/MooMed.Frontend/React/Modules/Account/Service/AccountValidationService.ts
// Functionality
import { IAccountValidationService } from "Definitions/Service"
import ModuleService from "modules/common/Service/ModuleService"
import * as accountValidationCommunication from "modules/Account/Communication/AccountValidationCommunication";

// Types
import { AccountValidationResult } from "modules/Account/types";

export default class AccountValidationService extends ModuleService implements IAccountValidationService {

	public constructor() {
		super()
	}

	public async start(): Promise<void> {

	}

	public async validateRegistration(accountId: number, token: string){
		const response = await accountValidationCommunication.validateRegistration(accountId, token);

		return {
			success: response.success,
			identityErrorCode: response.payload.identityErrorCode,
		} as AccountValidationResult;
	}
	
}

File: Services/MooMed.AccountValidationService/Service/AccountValidationService.cs
using System.Threading.Tasks;
using JetBrains.Annotations;
using MooMed.Common.Definitions.Models.User;
using MooMed.Common.Definitions.Models.User.ErrorCodes;
using MooMed.Common.ServiceBase.ServiceBase;
using MooMed.Core.DataTypes;
using MooMed.Logging.Abstractions.Interface;
using MooMed.Module.AccountValidation.Service.Interface;
using MooMed.ServiceBase.Services.Interface;

namespace MooMed.AccountValidationService.Service
{
	public class AccountValidationService : ServiceBaseWithLogger, IAccountValidationService
	{
		[NotNull]
		private readonly IEmailValidationService _emailValidationService;

		public AccountValidationService(
			[NotNull] IMooMedLogger logger,
			[NotNull] IEmailValidationService emailValidationService)
			: base(logger)
		{
			_emailValidationService = emailValidationService;
		}

		/// <summary>
		/// Takes care of the validation of an account
		/// </summary>
		/// <param name="accountValidationModel">Object containing accountId and token</param>
		/// <returns></returns>
		public async Task<ServiceResponse<IdentityErrorCode>> ValidateRegistration(AccountValidationModel accountValidationModel)
		{
			var validationResult = await _emailValidationService.ValidateAccount(accountValidationModel);

			return validationResult != IdentityErrorCode.Success
				? ServiceResponse.Success(validationResult)
				: ServiceResponse<IdentityErrorCode>.Failure();
		}
	}
}